### ISSUE TITLE
Port to Godot 3?

### ISSUE THREAD

**Issue by realkotob** [cyberfilth/Godot_Roguelike#106]
I ran the project with godot 2.1 and it works great, but I was wondering if you'd like to try porting it to godot 3?

I converted the project to 3.1.1 by exporting from the 2.1 editor, and I got a bunch of errors but they don't seem very serious, so would you be interested in merging the changes if I finish porting the project?

**Comment by cyberfilth** [comment]
I looked into exporting the project a while back and didn't think that it was worth it since Godot 2.1 handles 2D just fine, then the code grew and it seemed like too much work to port it over anyway.

At this point though, I think that development would be easier if it was using the latest version of Godot so, if you're up for it, I'd be more than happy to merge the changes.

**Comment by realkotob** [comment]
@cyberfilth I did a large part of the porting in https://github.com/cyberfilth/Ashgard-Keep/pull/107 and the game loads now, but there are still several issues to fix before it's up to date with the godot 2.1 version.

- [x] Game map is not visible.
- [x] Dungeon themes tile list.
- [x] Interface not clickable (e.g. main menu).

**Comment by realkotob** [comment]
Will add some todo items here as a sort of tracker list.

- [x]  Light cones don't seem to be working, maybe also item light (e.g. from potions).
- [ ]  Particle effects need to be remade.
- [x]  Attach the normal maps to their tiles.

**Comment by cyberfilth** [comment]
This is coming together really well, and really quickly too.
Nice work

**Comment by realkotob** [comment]
Thank you, this is not my first time porting godot projects unfortunately 😆 

I'm quite happy with the project so far, I just have to re-make the particle effects and figure out why the interface isn't responding to the mouse input and we should be at feature parity again. 

After that I'm going to have to start doing more granular testing to see if there are any corner cases/nuances/regressions that need fixing but I don't think there'll be a lot of those.

**Comment by cyberfilth** [comment]
I'm hitting a brick wall with the mouse input.
I've tried rebuilding the TitleMenu scene in the game but it didn't work, then I built a working menu in a standalone project but couldn't get it working once I ported it to the game.

**Comment by realkotob** [comment]
@cyberfilth Yeah sorry been busy with work, will do more debugging now and report on what works/doesn't.

**Comment by realkotob** [comment]
@cyberfilth That was some of the most intense debugging I've ever done #112  :b

I can spare some time to do the particle effects this weekend (I just have to open the editors side by side and recreate them), and after that we can do one last check to see if there's anything not ported properly before merging.

**Comment by cyberfilth** [comment]
@asheraryam So the transition scene was hijacking mouse input? that's a really good catch.
I feel kinda guilty that you've spent so much time making sense out of my hacky code whilst I've taken a 2 week break from adding new content (although I tell myself that it's so the 2 versions don't get out of sync!).

I should have some time this weekend to do some play-testing too so, all being well, I'll be happy to merge the branch.

**Comment by realkotob** [comment]
@cyberfilth Don't worry about it, I'm actually happy my work is being merged quickly without having to do rebasing on newer content conflicts.

It's also been a good educational experience, I always believe that the best way to understand a project is to be forced to go through all the dirty parts (looking at you android zelda-rpg that uses native rendering).
